state/txindex/kv/kv.go

```diff
@@ -180,9 +180,9 @@ func (txi *TxIndex) Search(q *query.Query) ([]*types.TxResult, error) {
 	} else if ok {
 		res, err := txi.Get(hash)
 		if res == nil {
```

Comment:
what do you think about using a switch here?

```go
switch {
	case err != nil:
		return []*types.TxResult{}, errors.Wrapf(err, "can't get tx %X", hash)
	case res == nil:
		return []*types.TxResult{}, nil
	default:
		return []*types.TxResult{res}, nil
```

Reply:
Good comment :) I think it's nice to have explicit error handling for each state.
I'll change to use the switch statement.